PerimeterXModule/PxCustomFunctions.cs

```diff
@@ -0,0 +1,151 @@
+                        customHandlerName));
+                }
+            }
+            catch (ReflectionTypeLoadException ex)
+            {
+                PxLoggingUtils.LogError(string.Format("Failed to load the ICustomLoginSuccessfulHandler '{0}': {1}.",
+                                              customHandlerName, ex.Message));
+            }
+            catch (Exception ex)
+            {
+                PxLoggingUtils.LogError(string.Format("Encountered an error while retrieving the ICustomLoginSuccessfulHandler '{0}': {1}.",
+                                              customHandlerName, ex.Message));
+            }
+
+            return null;
+        }
+
+        private static IEnumerable<Type> getAssebliesTypes()
+        {
+            return AppDomain.CurrentDomain.GetAssemblies()
+                    .SelectMany(a =>
+                    {
+                        try
+                        {
+                            return a.GetTypes();
+                        }
+                        catch
+                        {
+                            return new Type[0];
+                        }
+                    });
+        }
+
+        /// <summary>
+        /// Uses reflection to check whether an IVerificationHandler was implemented by the customer.
+        /// </summary>
+        /// <returns>If found, returns the IVerificationHandler class instance. Otherwise, returns null.</returns>
+        public static IVerificationHandler GetCustomVerificationHandler(string customHandlerName)
+		{
+			if (string.IsNullOrEmpty(customHandlerName))
+			{
+				return null;
+			}
+
+			try
+			{
+				var customVerificationHandlerType = getAssebliesTypes()
+							 .FirstOrDefault(t => t.GetInterface(typeof(IVerificationHandler).Name) != null &&
+												  t.Name.Equals(customHandlerName) && t.IsClass && !t.IsAbstract);
+
+				if (customVerificationHandlerType != null)
+				{
+					var instance = (IVerificationHandler)Activator.CreateInstance(customVerificationHandlerType, null);
+					PxLoggingUtils.LogDebug(string.Format("Successfully loaded ICustomeVerificationHandler '{0}'.", customHandlerName));
+					return instance;
+				}
+				else
+				{
+					PxLoggingUtils.LogDebug(string.Format(
+						"Missing implementation of the configured IVerificationHandler ('customVerificationHandler' attribute): {0}.",
+						customHandlerName));
+				}
+			}
+			catch (ReflectionTypeLoadException ex)
+			{
+				PxLoggingUtils.LogError(string.Format("Failed to load the ICustomeVerificationHandler '{0}': {1}.",
+											  customHandlerName, ex.Message));
+			}
+			catch (Exception ex)
+			{
+				PxLoggingUtils.LogError(string.Format("Encountered an error while retrieving the ICustomeVerificationHandler '{0}': {1}.",
+											  customHandlerName, ex.Message));
+			}
+
+			return null;
+		}        
```

Comment:
Lots of this seems repetitive for all these custom functions - perhaps things like the null checks, try/catch can be separated into a more generic function?